app/javascript/dashboard/components/buttons/ResolveAction.vue

```diff
@@ -1,149 +1,151 @@
-<script>
-import { mapGetters } from 'vuex';
+<script setup>
+import { ref, computed } from 'vue';
 import { useAlert } from 'dashboard/composables';
-import keyboardEventListenerMixins from 'shared/mixins/keyboardEventListenerMixins';
+import { useI18n } from 'dashboard/composables/useI18n';
+import { useStore, useStoreGetters } from 'dashboard/composables/store';
+import { useEmitter } from 'dashboard/composables/emitter';
+import { useKeyboardEvents } from 'dashboard/composables/useKeyboardEvents';
+
 import WootDropdownItem from 'shared/components/ui/dropdown/DropdownItem.vue';
 import WootDropdownMenu from 'shared/components/ui/dropdown/DropdownMenu.vue';
-
 import wootConstants from 'dashboard/constants/globals';
 import {
   CMD_REOPEN_CONVERSATION,
   CMD_RESOLVE_CONVERSATION,
-} from '../../routes/dashboard/commands/commandBarBusEvents';
+} from 'dashboard/routes/dashboard/commands/commandBarBusEvents';
 
-export default {
-  components: {
-    WootDropdownItem,
-    WootDropdownMenu,
-  },
-  mixins: [keyboardEventListenerMixins],
-  data() {
-    return {
-      isLoading: false,
-      showActionsDropdown: false,
-      STATUS_TYPE: wootConstants.STATUS_TYPE,
-    };
+const store = useStore();
+const getters = useStoreGetters();
+const { t } = useI18n();
+
+const resolveActionsRef = ref(null);
+const arrowDownButtonRef = ref(null);
+const isLoading = ref(false);
+const showActionsDropdown = ref(false);
+
+const currentChat = computed(() => getters.getSelectedChat.value);
+
+const isOpen = computed(
+  () => currentChat.value.status === wootConstants.STATUS_TYPE.OPEN
+);
+const isPending = computed(
+  () => currentChat.value.status === wootConstants.STATUS_TYPE.PENDING
+);
+const isResolved = computed(
+  () => currentChat.value.status === wootConstants.STATUS_TYPE.RESOLVED
+);
+const isSnoozed = computed(
+  () => currentChat.value.status === wootConstants.STATUS_TYPE.SNOOZED
+);
```

Comment:
> [!NOTE]
> No need to implement this now. Just keep in mind for future enhancements

Are we using similar pattern elsewhere? Where we access the currentChat and it's status

We can use a composable `useCurrentChat`

```
import { computed } from 'vue';
import { useStoreGetters } from 'dashboard/composables/store';
import wootConstants from 'dashboard/constants/globals';

export const useCurrentChat = () => {
  const getters = useStoreGetters();

  const currentChat = computed(() => getters.getSelectedChat.value);

  const isOpen = computed(
    () => currentChat.value.status === wootConstants.STATUS_TYPE.OPEN
  );
  const isPending = computed(
    () => currentChat.value.status === wootConstants.STATUS_TYPE.PENDING
  );
  const isResolved = computed(
    () => currentChat.value.status === wootConstants.STATUS_TYPE.RESOLVED
  );
  const isSnoozed = computed(
    () => currentChat.value.status === wootConstants.STATUS_TYPE.SNOOZED
  );

  return {
    currentChat,
    isOpen,
    isPending,
    isResolved,
    isSnoozed,
  };
};
```

Reply:
Noted. I just checked the code base and saw that only using in two files, one is this file and other is `ConversationHeader.vue`